sia/src/rhp/quic.rs

```diff
@@ -260,13 +236,68 @@ impl Dialer {
         self.set_cached_prices(&host_key, prices.clone());
         Ok(prices)
     }
+}
+
+#[derive(Debug)]
+struct DialerInner {
+    /*
+        note: quinn's documentation (https://docs.rs/quinn/latest/quinn/struct.Endpoint.html#method.client) suggests
+        non-ideal fallback behavior when dual-stack is not supported. This effectively treats every platform as
+        single-stack instead since IPv4 is the preferred fallback.
+    */
+    endpoint_v4: Option<Endpoint>,
+    endpoint_v6: Option<Endpoint>,
+
+    hosts: Mutex<HashMap<PublicKey, Vec<NetAddress>>>,
+    open_conns: Mutex<HashMap<PublicKey, Connection>>,
+    cached_prices: Mutex<HashMap<PublicKey, HostPrices>>,
+}
+
+impl DialerInner {
+    fn new(mut client_config: ClientConfig) -> Result<Self, &'static str> {
```

Comment:
These `Result<Self, &'static str>` are convenient but it makes it means that a caller can't easily use `?` here after implementing `From<Error>` for their error type. Might be better to add an additional
```rust
#[error("unable to create IPv4 and IPv6 endpoint"]
NoEndpoint,
```
to the `Error` type of this module.